frontend/packages/dev-console/src/components/topology/__tests__/graph-test-data.ts

```diff
@@ -60,7 +60,6 @@ export const MockGraphResources: TopologyDataResources = {
           replicas: 1,
           revisionHistoryLimit: 10,
           test: false,
-          selector: { name: 'cakephp-mysql-example' },
```

Comment:
These selectors are valid for deployment configs. They just aren't type `Selector`.

See my change in #3017

Reply:
Newer resources like `ReplicaSet` use the new format of selector described by type `Selector`. Older resources use a generic name/value map.

Reply:
We'll eventually have a `DeploymentConfigKind` with

```ts
selector: MatchLabels;
```

and a `ReplicaSetKind` with

```ts
selector: Selector;
```

which is why I'd rather not overload what the `Selector` types means.

I'd like to phase out `K8sResourceKind` in the long run in favor of specific type declarations. For unknown types, we can have a more restrictive type like `UnknownK8sResource` that only allows the common properties we always expect like object metadata.

The project is large enough that I think we'd really benefit from better type checking. We've caught bugs already as we've tightened down the types we have.

Reply:
@spadgett Yes, I agree with your comments, these selectors are valid just not type `Selector`. I removed these because they are mock data and selector is not being used in the tests specifically.

Reply:
But they're required for deployment configs. When we add a `DeploymentConfigKind` type, we're going to have to add these back. It's better to leave these as-is and go with the change in #3017.